src/main/java/io/tus/java/client/TusUploader.java

```diff
@@ -280,6 +280,30 @@ public void finish() throws ProtocolException, IOException {
         // that we will not need to read from it again in the future.
         input.close();
     }
+    
+    /**
+     * Finish the request by closing the HTTP connection. You can choose whether to close the Input Stream or not.
+     * You can call this method even before the entire file has been uploaded.
+     * It pauses the upload properly to be resumed in future.
+     * Be aware it doesn't release local resources as it does not close the File's
+     * Input Stream if {@code closeStream == false}
+     * To be safe use {@link TusUploader#finish()}.
+     * @param closeInputStream Determines whether the InputStream is closed with the HTTP connection. Not closing the
+     *                         Input Stream may be useful for future upload a future continuation of the upload.
+     * @throws ProtocolException Thrown if the server sends an unexpected status code
+     * @throws IOException  Thrown if an exception occurs while cleaning up.
+     */
+    public void finish(boolean closeInputStream) throws ProtocolException, IOException {
+        if (closeInputStream){
+            finish();
+        } else {
+            finishConnection();
+            if (upload.getSize() == offset) {
+                client.uploadFinished(upload);
+            }
+            input.seekTo(0);
```

Comment:
I don't think this operation is possible at all times, in particular when we are uploading from a streaming resource and not a file-based input stream. In such cases it is not possible to seek to the beginning of the stream, as the data is already erased from process.

In general, we cannot know here which offset to seek to, as we need to ask the server first, how much data it received. However, the current TusUploader class is not designed to send those HEAD requests. Frankly, I am not sure how to solve this situation right now, but it's probably best to discuss this in a call :)